module/src/main/java/zipkin2/module/storage/forwarder/ZipkinForwarderStorageModule.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2019-2020 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package zipkin2.module.storage.forwarder;
+
+import org.springframework.boot.autoconfigure.condition.ConditionalOnBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import zipkin2.reporter.Sender;
+import zipkin2.reporter.kafka.KafkaSender;
+import zipkin2.reporter.okhttp3.OkHttpSender;
+import zipkin2.storage.StorageComponent;
+import zipkin2.storage.forwarder.ForwarderStorage;
+
+@Configuration
+@EnableConfigurationProperties({
+  ZipkinHttpForwarderStorageProperties.class,
+  ZipkinKafkaForwarderStorageProperties.class,
+})
+@ConditionalOnMissingBean(StorageComponent.class)
+public class ZipkinForwarderStorageModule {
```

Comment:
you'll notice the module is combined now. this is actually much easier to deal with than separate docker images: spring conditions can already dispatch on properties.